### ISSUE TITLE
clarify behavior of --lr_scheduler in LightningCLI

### ISSUE THREAD

**Issue by kaczmarj** [Lightning-AI/lightning#17547]
### 📚 Documentation

hello, wanted to say that i love using lightning. i am currently using the `LightningCLI` object to train my models. i implemented `configure_optimizers` on my LightningModule subclass, and using the cli, i was hoping to change the learning rate scheduler but use the optimizer defined by my `configure_optimizers`. in other words, i was only passing `--lr_scheduler` and not `--optimizer`. after some playing around with parameters and browsing the source, i see that `--lr_scheduler` only has an effect if `--optimizer` is also passed.

would you be interested in a pr that clarifies the `--help` text that `--lr_scheduler` requires `--optimizer`?

cc @borda @carmocca @mauvilsa

**Comment by awaelchli** [comment]
It sounds good to me. Maybe it could be added in this section https://pytorch-lightning.readthedocs.io/en/1.6.5/common/lightning_cli.html#optimizers-and-learning-rate-schedulers
if @mauvilsa @carmocca agree.

**Comment by mauvilsa** [comment]
> Maybe it could be added in this section https://pytorch-lightning.readthedocs.io/en/1.6.5/common/lightning_cli.html#optimizers-and-learning-rate-schedulers

That is a link to a very old version of the documentation. Now this would be improving https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_intermediate_2.html#multiple-schedulers or https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_advanced_3.html#multiple-optimizers-and-schedulers. Indeed this could be explained better.

@kaczmarj didn't you get warnings? In the second link you can see that the CLI automatically implements `configure_optimizers` which means your implementation of it would be ignored and I guess a warning shown. The automatic `configure_optimizers` only supports the most basic case, one optimizer and optionally one scheduler. For any other case the automatic `configure_optimizers` should not be used.

**Comment by kaczmarj** [comment]
i did not get any warnings but it would have been nice ;) the code must have gone through this path (below), where the function returns early without a warning if `--optimizer` is not given.

https://github.com/Lightning-AI/lightning/blob/9eedf7ae4b5cd398931af14ca7eac0605ef740ee/src/lightning/pytorch/cli.py#L586-L590

i opened a pr for this issue #17552 that updates https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_intermediate_2.html#multiple-schedulers
